#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("task1_Kp_pos", double_t, 0, "task1_Kp_pos", 100, 0, 600)#
gen.add("task1_Kp_ori", double_t, 0, "task1_Kp_ori", 5,  0, 30)#

gen.add("task1_Kv_pos", double_t, 0, "task1_Kv_pos", 10,  0, 90)#
gen.add("task1_Kv_ori", double_t, 0, "task1_Kv_ori", 0.5,   0, 18)#

gen.add("task2_Md", double_t, 0, "inertia", 3, 0, 100)#
gen.add("task2_Bd", double_t, 0, "damping", 8, 0, 100)
gen.add("task2_Kd", double_t, 0, "stiffness", 4, 0, 100)
#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "nullspace_impedance_controller_param"))
